File: matiasmasso/Mat.NET/DTO/DTOMeeting.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOMeeting
    {
        public Guid Guid { get; set; }
        public DateTime Fch { get; set; }
        public DTOContact Place { get; set; }
        public Medias Media { get; set; }
        public string Subject { get; set; }
        public List<DTOUser> Presents { get; set; }
        public DTOUser UsrCreated { get; set; }
        public string PresentOthers { get; set; }
        public List<DTOMeetingStatement> Statements { get; set; }

        public bool IsNew { get; set; }
        public bool IsLoaded { get; set; }

        public enum Medias
        {
            NotSet,
            Presencial,
            VideoCall
        }
    }

    public class DTOMeetingStatement
    {
        public Guid Guid { get; set; }
        public int Ord { get; set; }
        public string Subject { get; set; }
        public string Body { get; set; }
        public List<DTOMeetingTask> Tasks { get; set; }
    }

    public class DTOMeetingTask
    {
        public Guid Guid { get; set; }
        public string Task { get; set; }
        public DateTime Deadline { get; set; }
        public DTOUser User { get; set; }
        public StatusEnum Status { get; set; }

        public enum StatusEnum
        {
            Open,
            EnCurso,
            Done
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOMeeting.vb
Public Class DTOMeeting
    Property Guid As Guid
    Property Fch As Date
    Property Place As DTOContact
    Property Media As Medias
    Property Subject As String
    Property Presents As List(Of DTOUser)
    Property UsrCreated As DTOUser
    Property PresentOthers As String
    Property Statements As List(Of DTOMeetingStatement)

    Property IsNew As Boolean
    Property IsLoaded As Boolean

    Public Enum Medias
        NotSet
        Presencial
        VideoCall
    End Enum
End Class

Public Class DTOMeetingStatement
    Property Guid As Guid
    Property Ord As Integer
    Property Subject As String
    Property Body As String
    Property Tasks As List(Of DTOMeetingTask)
End Class

Public Class DTOMeetingTask
    Property Guid As Guid
    Property Task As String
    Property Deadline As Date
    Property User As DTOUser
    Property Status As StatusEnum

    Public Enum StatusEnum
        Open
        EnCurso
        Done
    End Enum

End Class


